scripts/setup_git_hooks

```diff
@@ -1,8 +1,6 @@
 #!/bin/bash
 # sets up pre commit hooks
```

Comment:
It seems like this script was never used - not sure why - but now it is.

Reply:
One sort of related thing with `clang_format_git_diff` now being used in `setup_git_hooks`: if the engineer's Homebrew configuration is non-standard (such that it isn't symlink installations into `/usr/local/bin`), `clang_format_git_diff` will fail unexpectedly. Frankly I'm not even positive if that's a configurable Homebrew thing or not, but might be safer to replace `/usr/local/bin/clang-format` with just `clang-format` as the `clang_format_all` script does.